src/opera/template/topology.py

```diff
@@ -68,3 +68,117 @@ def get_artifact(self, params):
         node_name, *rest = params
 
         return self.find_node(node_name).get_artifact(["SELF"] + rest)
+
+    def concat(self, params):
+        if not isinstance(params, list):
+            raise DataError("Concat intrinsic function parameters for '{}'"
```

Comment:
If user provides wrong parameter for **join** function, e.g something like 
```yaml
 join: [ [ "t", "o", "s", "c", "a", { get_input: input, get_input: input } ] ] 
```
the error would be
```
Dict value expression for concat function has too many keys: { get_input: input, get_input: input }
```
Which might be confusing as **concat** function is not used

Reply:
I agree, this may confuse someone. However, I don't see it as a problem since user used two elements inside the dict value expression, which as expected resulted in error. I didn't see an example in TOSCA that uses concat or join intrinsic function with dict that has multiple elements. Or have I missed something? What do you suggest here?

Reply:
Sorry if it was confusing. I just meant that user used **join** function but got error for **concat** which might be misleading, since **concat** may be not used at all. The sample is just the piece of TOSCA I used to test the error.

Reply:
Ahaa I get it now :smiley:. I have corrected this part in join so it would say: `Dict value expression for concat or join function has too many keys...`.